homeassistant/components/backup/util.py

```diff
@@ -117,6 +118,12 @@ def read_backup(backup_path: Path) -> AgentBackup:
         )
 
 
+def suggested_filename(backup: AgentBackup) -> str:
+    """Suggest a filename for the backup."""
+    date = dt_util.parse_datetime(backup.date, raise_on_error=True)
+    return f"{backup.name} - {date.strftime('%Y-%m-%d %H.%M %S%f')}.tar"
```

Comment:
Hm, so the ID part is just part of the timestamp, I think I'd then just make it part of the timestamp, as in:

```suggestion
    return f"{backup.name} - {date.strftime('%Y-%m-%d %H-%M-%S.%f')}.tar"
```

Reply:
`'%Y-%m-%d %H:%M:%S.%f'`?

Reply:
Can we have underscore instead of space? I loathe filenames with spaces.

Reply:
I saw a note in the beta channel that not all file systems like the `:` character. 

Not sure if that's true, but I thought it would be worth leaving this note here

Reply:
: in the filename is very likely an issue for OneDrive

Reply:
> Can we have underscore instead of space? I loathe filenames with spaces.

OK, but it's also much less readable.

> '%Y-%m-%d %H:%M:%S.%f'

Not with the actual colon character `:`, as it doesn't work on most Windows filesystems, including NTFS. What we *could* do is some alternative look-alike for example [U+2236](https://www.compart.com/en/unicode/U+2236) (`∶`), but I'm not sure if that will display correctly in all cases, and it's also very confusing because you won't be able to type the filename unless (unless you can tell `:` from `∶`, and know how to type the latter).

Reply:
I'd just go with a dash `-` instead -  the intent is to make the name more readable/understandable for humans instead of a hash. Even with a dash, it's already way better than the current filename.

Most files will get deleted without anyone ever seeing the filename at all, so I guess that a time with dashes in it is pretty much acceptable for the few that need to restore something/need to look up the file.

Reply:
> so the ID part is just part of the timestamp, I think I'd then just make it part of the timestamp

We tried that, and it makes the timestamp harder to read, and look like a date:
`Custom backup 2025.1.0 - 2025-01-30 05-42-12.345678.tar`
vs:
`Custom backup 2025.1.0 - 2025-01-30 05.42 12345678.tar`

Reply:
<sub>Personally I like the current state date-time formatting. Perfectly understandable by humans.</sub>

Reply:
It's the current state *of this PR*. The current state on dev is `{backup_id}.tar` which is extremely unfriendly to the user because the `backup_id` is not visible to users.

Reply:
I'd remove the dash between underscores when using underscores to separate words.

The point with underscores is that it will make the filename more readable and identifyable in contexts where spaces are not always visible.

Reply:
>  underscores .. make the filename more readable and identifyable in contexts where spaces are not always visible

Can you please give examples of such contexts, and actual problems it creates?

Reply:
2 spaces next to each other is hard to distinguish from one space if all characters are not shown with equal width which is common in many contexts.

A leading or ending space in the filename can be invisible depending on the context.

Reply:
Those are valid points. I think we should fix that by trimming the backup name though.

Reply:
It's not just about how we name the file. If we don't include any spaces in the default name, the user will be less inclined to introduce any spaces later unless they completely rename the file. Hopefully this will lead to a better experience for users with less confusion.

Reply:
Why would we want to stop users from introducing spaces in their filenames?

Reply:
Because it increases the risk for mistakes and confusion per above.

Reply:
That's a circular argument though

Reply:
No, spaces in filenames are intrinsically less visible than other characters making it more likely to not identify them correctly.